runtime/lua/vim/ui.lua

```diff
@@ -139,12 +144,17 @@ function M.open(path)
     path = vim.fs.normalize(path)
   end
 
-  local cmd --- @type string[]
+  local cmd --- @type string[]|string[][]
   local opts --- @type vim.SystemOpts
 
   opts = { text = true, detach = true }
 
-  if vim.fn.has('mac') == 1 then
+  if cmdpath then
+    -- not checking for the existence of the cmd command, because
+    -- it could be a more complicated string or even list of
+    -- string, not just the name of the executable
+    cmd = { cmdpath, path }
```

Comment:
if cmdpath is a list then this won't work. rather than supporting `string|string[]`, let's require `string[]`.

```suggestion
    cmd = vim.list_extend(cmdpath, { path })
```